### PR TITLE
Add usage of Composite

### PR DESCRIPTION
**Pull request by sgligorijevicTT**
### Ticket
Related to #618

### What's changed
Code was added to monkeypatch jax to use StableHLO CompositeOp for Gelu from our tests, this should bring a speedup and cause more models to pass with good accuracy as we are now avoiding a known bug in metal.

### Checklist
- [X] New/Existing tests provide coverage for changes

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -3,6 +3,8 @@
 # SPDX-License-Identifier: Apache-2.0
 
 from contextlib import contextmanager
+from functools import partial
```
Comment (ajakovljevicTT) [inline tests/conftest.py:6]: nit: please sor in alphabetical order

File: tests/conftest.py [hunk 2]
```diff
@@ -244,3 +246,41 @@ def initialize_device_connectors():
...
     DeviceConnectorFactory.create_connector(Framework.JAX)
     DeviceConnectorFactory.create_connector(Framework.TORCH)
+
+
+real_gelu = None
+
+
+@pytest.fixture(autouse=True)
+def monkeypatch_import(request):
+    import jax
```
Comment (ajakovljevicTT) [inline tests/conftest.py:256]: put imports (when needed, on top)

File: tests/conftest.py [hunk 3]
```diff
@@ -244,3 +246,41 @@ def initialize_device_connectors():
...
+            )(x)
+        else:
+            return jax.lax.composite(
+                lambda x: real_gelu(x, approximate=False),
+                "tt.gelu",
+            )(x)
+
+    jax.nn.gelu = mygelu
+
+    import transformers
```
Comment (ajakovljevicTT) [inline tests/conftest.py:277]: same here

File: tests/conftest.py [hunk 4]
```diff
@@ -244,3 +246,41 @@ def initialize_device_connectors():
...
+    import transformers.modeling_flax_utils
+
+    transformers.modeling_flax_utils.ACT2FN["gelu"] = partial(
+        mygelu, approximate=False
+    )  # I am not sure why I need to patch ACT2FN too when it pulls jax.nn.gelu
```
Comment (ajakovljevicTT) [inline tests/conftest.py:282]: leave a nicely formatted TODO with an issue attached

File: tests/conftest.py [hunk 5]
```diff
@@ -244,3 +246,41 @@ def initialize_device_connectors():
...
+@pytest.fixture(autouse=True)
+def monkeypatch_import(request):
+    import jax
+
+    global real_gelu
+    if real_gelu is None:
+
+        real_gelu = jax.nn.gelu
+
+    def mygelu(x, approximate=True):
```
Comment (ajakovljevicTT) [inline tests/conftest.py:263]: nit: I think this function could use a better name, something like "custom_tt_gelu" (or similar)

File: tests/conftest.py [hunk 6]
```diff
@@ -244,3 +246,41 @@ def initialize_device_connectors():
     """
     DeviceConnectorFactory.create_connector(Framework.JAX)
     DeviceConnectorFactory.create_connector(Framework.TORCH)
+
+
+real_gelu = None
+
+
+@pytest.fixture(autouse=True)
+def monkeypatch_import(request):
```
Comment (ajakovljevicTT) [inline tests/conftest.py:255]: leave a comment explaining what this function does

File: tests/conftest.py [hunk 7]
```diff
@@ -244,3 +246,41 @@ def initialize_device_connectors():
     """
     DeviceConnectorFactory.create_connector(Framework.JAX)
     DeviceConnectorFactory.create_connector(Framework.TORCH)
+
+
+real_gelu = None
```
[the remark this answers is not in the document]
Comment (sgligorijevicTT) [reply tests/conftest.py]: I pushed a refactor which should make everything nicer

**Review by ajakovljevicTT — APPROVED** [review]
